apps/admin_api/lib/admin_api/v1/controllers/category_controller.ex

```diff
@@ -0,0 +1,122 @@
+defmodule AdminAPI.V1.CategoryController do
+  use AdminAPI, :controller
+  import AdminAPI.V1.ErrorHandler
+  alias EWallet.CategoryPolicy
+  alias EWallet.Web.{SearchParser, SortParser, Paginator}
+  alias EWalletDB.Category
+
+  # The field names to be mapped into DB column names.
+  # The keys and values must be strings as this is mapped early before
+  # any operations are done on the field names. For example:
+  # `"request_field_name" => "db_column_name"`
+  @mapped_fields %{
+    "created_at" => "inserted_at"
+  }
+
+  # The fields that are allowed to be searched.
+  # Note that these values here *must be the DB column names*
+  # If the request provides different names, map it via `@mapped_fields` first.
+  @search_fields [:id, :name, :description]
+
+  # The fields that are allowed to be sorted.
+  # Note that the values here *must be the DB column names*.
+  # If the request provides different names, map it via `@mapped_fields` first.
+  @sort_fields [:id, :name, :description, :inserted_at, :updated_at]
+
+  defp permit(action, user_id, category_id) do
+    Bodyguard.permit(CategoryPolicy, action, user_id, category_id)
+  end
+
+  @doc """
+  Retrieves a list of categories.
+  """
+  def all(conn, attrs) do
+    with :ok <- permit(:all, conn.assigns.user.id, nil) do
+      categories =
+        Category
+        |> SearchParser.to_query(attrs, @search_fields, @mapped_fields)
+        |> SortParser.to_query(attrs, @sort_fields, @mapped_fields)
+        |> Paginator.paginate_attrs(attrs)
+
+      case categories do
```

Comment:
No `respond_multiple` / `respond_single`? 🤔

Reply:
The controllers started to have branching and the `respond_multiple` / `respond_single` are difficult to handle now.

Will do a separate PR that creates macro like `if_allowed user, :get, object_id, action do ... end` instead.